demuxer.go

```diff
@@ -27,7 +27,7 @@ type Demuxer struct {
 	optPacketsParser PacketsParser
 	packetBuffer     *packetBuffer
 	packetPool       *packetPool
-	programMap       programMap
+	programMap       *programMap
```

Comment:
My understanding is that in GO, maps are passed by reference therefore you shouldn't need to use a pointer here. Did you add the pointer by necessity or because you felt it was the right move?

Reply:
You're right, but programMap here is a struct which contains a map. It felt safer to copy by reference in case other fields were added in the future.

Reply:
👍